win32ss/user/ntuser/ghost.c

```diff
@@ -128,6 +219,109 @@ HWND FASTCALL UserHungWindowFromGhostWindow(HWND hwndGhost)
     return IntHungWindowFromGhostWindow(pGhostWnd);
 }
 
+static BOOL IntHaveToQuitGhosting(void)
+{
+    return !!NtUserFindWindowEx(NULL, NULL, &GhostClass, NULL, 0);
+}
+
+static ULONG NTAPI
+GhostThreadProc(PVOID Param)
+{
+    HWND hwndTarget = (HWND)Param;
+    MSG msg;
+
+    if (!IntCreateGhostWindow(hwndTarget))
+    {
+        DPRINT1("Unable to create ghost\n");
+        goto Quit;
+    }
+
+    // thread message loop
+    while (NtUserGetMessage(&msg, NULL, 0, 0))
+    {
+        switch (msg.message)
+        {
+            case PM_CREATE_GHOST:
+            {
+                hwndTarget = (HWND)msg.wParam;
+                if (!IntCreateGhostWindow(hwndTarget))
+                {
+                    DPRINT1("Unable to create ghost\n");
+                }
+                break;
+            }
+        }
+        NtUserTranslateMessage(&msg, 0);
```

Comment:
Same as above. Remember the rule: don't call NtUser functions from kernel mode.

Reply:
OK.